From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tariq Bashir <tbashir@brightsign.biz>
Date: Wed, 2 Jul 2025 11:02:18 +0100
Subject: BrightSign StoragePartitionConfig: Revert Modernize equality
 operators change

Chromium added https://chromium-review.googlesource.com/c/chromium/src/+/6554130
which modernized the equality operators in StoragePartitionConfig and other places.
This change means that the quota std::optional<int> we added is now included in the
equality and comparison operators, which is not what we want. Revert this change
to restore the previous behavior only for StoragePartitionConfig.

diff --git a/content/public/browser/storage_partition_config.cc b/content/public/browser/storage_partition_config.cc
index ff56aedfc7361645be5968f7b017161d999c149f..570f321bae270056420d713cb5cf5445df006b04 100644
--- a/content/public/browser/storage_partition_config.cc
+++ b/content/public/browser/storage_partition_config.cc
@@ -65,6 +65,34 @@ StoragePartitionConfig::GetFallbackForBlobUrls() const {
           quota_size_);
 }
 
+bool StoragePartitionConfig::operator<(
+    const StoragePartitionConfig& rhs) const {
+  if (partition_domain_ != rhs.partition_domain_)
+    return partition_domain_ < rhs.partition_domain_;
+  if (partition_name_ != rhs.partition_name_)
+    return partition_name_ < rhs.partition_name_;
+  if (in_memory_ != rhs.in_memory_)
+    return in_memory_ < rhs.in_memory_;
+  if (fallback_to_partition_domain_for_blob_urls_ !=
+      rhs.fallback_to_partition_domain_for_blob_urls_) {
+    return fallback_to_partition_domain_for_blob_urls_ <
+           rhs.fallback_to_partition_domain_for_blob_urls_;
+  }
+  return false;
+}
+bool StoragePartitionConfig::operator==(
+    const StoragePartitionConfig& rhs) const {
+  return partition_domain_ == rhs.partition_domain_ &&
+         partition_name_ == rhs.partition_name_ &&
+         in_memory_ == rhs.in_memory_ &&
+         fallback_to_partition_domain_for_blob_urls_ ==
+             rhs.fallback_to_partition_domain_for_blob_urls_;
+}
+bool StoragePartitionConfig::operator!=(
+    const StoragePartitionConfig& rhs) const {
+  return !(*this == rhs);
+}
+
 std::ostream& operator<<(std::ostream& out,
                          const StoragePartitionConfig& config) {
   out << "{";
diff --git a/content/public/browser/storage_partition_config.h b/content/public/browser/storage_partition_config.h
index b1b708c1982a28f40ee961d363f6b2583619f8f0..fb2970297615376c0c8b67ef68815f4df9dfa847 100644
--- a/content/public/browser/storage_partition_config.h
+++ b/content/public/browser/storage_partition_config.h
@@ -84,10 +84,9 @@ class CONTENT_EXPORT StoragePartitionConfig {
   }
   std::optional<StoragePartitionConfig> GetFallbackForBlobUrls() const;
 
-  friend bool operator==(const StoragePartitionConfig&,
-                         const StoragePartitionConfig&) = default;
-  friend auto operator<=>(const StoragePartitionConfig&,
-                          const StoragePartitionConfig&) = default;
+  bool operator<(const StoragePartitionConfig& rhs) const;
+  bool operator==(const StoragePartitionConfig& rhs) const;
+  bool operator!=(const StoragePartitionConfig& rhs) const;
 
  private:
   friend StoragePartitionConfig CreateStoragePartitionConfigForTesting(
